src/FluentNHibernate/PersistenceModel.cs

```diff
@@ -77,12 +77,15 @@ public void AddMappingsFromAssembly(Assembly assembly)
 
         public void AddMappingsFromSource(ITypeSource source)
         {
-            source.GetTypes()
+            foreach (object o in source.GetTypes()
                 .Where(x => IsMappingOf<IMappingProvider>(x) ||
                             IsMappingOf<IIndeterminateSubclassMappingProvider>(x) ||
                             IsMappingOf<IExternalComponentMappingProvider>(x) ||
                             IsMappingOf<IFilterDefinition>(x))
-                .Each(Add);
+                .AsParallel().Select(t => t.InstantiateUsingParameterlessConstructor()))

```

Comment:
Sometimes the order of mappings does matter. And this could break it.

Reply:
Thanks for reviewing.  Pushed up another change so the iteration of the parallel will maintain order.  Should still have the speed benefit of multi core but will aggregate the same way it originally was.